goss-testing/scripts/data-validator.py

```diff
@@ -192,103 +206,98 @@ def are_hosts_sane(data, desired_keys):
-  err = 0
-
-  lengths = []
-  counts = []
-  for blob in node_class:
-    lengths.append(blob['params-length'])
-  
-  for i in lengths:
-    occ = lengths.count(i)
-    counts.append(occ)
-    for blob in node_class:
-      if i == blob['params-length']:
-        blob['occurrence'] = occ
-
-  baseline = max(counts)
-
-  searches = [ "bond=bond0:mgmt0,mgmt1",
-               "ifname=mgmt0",
-               "ip=mgmt0",
-               "ifname=mgmt1",
-               "ip=mgmt1",
-               "hostname=" ]
-
-  all_param_strings = []
-  for blob in node_class:
-    for k,v in blob.items():
-      if k == "params-stripd":
-        all_param_strings.append(v)
-    if blob['occurrence'] == baseline:
-      for search in searches:
-        if re.search(search, blob['params-orig']):
-          blob['valid'] = True
-        else:
-          blob['valid'] = False
-    else:
-      blob['valid'] = False
-
-  unique_param_strings = set(all_param_strings)
-  if len(unique_param_strings) > 1:
-    for blob in node_class:
-      for params in unique_param_strings:
-        if params == blob['params-stripd']:
-          blob['valid'] = False
-
-
-  for blob in node_class:
-    if blob['valid'] == False:
-      print_err(f"WARN: {blob['hostname']}: has boot params that differ from at least one other node\n{blob['params-orig']}\n")
-      err = 1
-
-  print_err("------------------------------------------------------------")
-
-  if err == 1:
-    return err
-
-
 def boot_params(data):
-  """Attempt to validate boot parameters from BSS
+  """Confirm that params required for booting are
+     present.
   """
-  worker = []
-  storage = []
-  management = []
-  err = 0
 
-  macreg = re.compile(r':([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}')
-  hostreg = re.compile(r'hostname=[a-zA-Z]*\-[a-zA-Z0-9]*')
-  httpreg = re.compile(r's=http://.*/')
+  err = 0
+  ncn_params = [
+	"biosdevname=1",
+	"console=tty0",
+	"console=ttyS0,115200",
+	"crashkernel=[0-9]*M",
+	"ds=nocloud-net;s=http://",
+	"hostname=",
+	"ifname=mgmt0",
+	"ifname=mgmt1",
+	"initrd=initrd.img.xz",
+	"iommu=pt",
+	"log_buf_len=1",
+	"metal.no-wipe=",
+	"metal.no-wipe=1",
+	"metal.server=http://",
+	"pcie_ports=native",
+	"psi=1",
+	"rd.auto=1",
+	"rd.bootif=0",
+	"rd.dm=0",
+	"rd.live.overlay.overlayfs=1",
+	"rd.live.overlay.thin=1",
+	"rd.live.overlay=LABEL=ROOTRAID",
+	"rd.live.ram=0",
+	"rd.live.squashimg=filesystem.squashfs",
+	"rd.luks.crypttab=0",
+	"rd.lvm.conf=0",
+	"rd.lvm=1",
+	"rd.md.conf=1",
+	"rd.md.waitclean=1",
+	"rd.md=1",
+	"rd.multipath=0",
+	"rd.neednet=0",
+	"rd.net.dhcp.retry=5",
+	"rd.net.timeout.carrier=120",
+	"rd.net.timeout.iflink=120",
+	"rd.net.timeout.ifup=120",
+	"rd.net.timeout.ipv6auto=0",
+	"rd.net.timeout.ipv6dad=0",
+	"rd.peerdns=0",
+	"rd.retry=10",
+	"rd.shell",
+	"rd.skipfsck",
+	"rd.writable.fsimg=0",
+	"root=live:LABEL=SQFSRAID",
+	"rootfallback=LABEL=BOOTRAID",
+	"transparent_hugepage=never"
+	]
```

Comment:
The most important ones are these:

- `transparent_hugepage=never`
- `rd.peerdns=0`
- `rd.multipath=0`
- `psi=1`
- `rd.live.*`
- `rd.md.waitclean=1`
- `biosdevname=1`
- `ifname=mgmt*`
- `rd.luks{,=0}`

The rest are important, but the ones listed above will cause the node to boot into a very undesirable state if they're not set properly.

Reply:
So should I only look for the items in your bulleted list? Like, that's the comprehensive list?